src/KafkaFlow/Serializer/IDeserializer.cs

```diff
@@ -0,0 +1,21 @@
+namespace KafkaFlow.Serializer
+{
+    using System;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Used to implement a message serializer
+    /// </summary>
+    public interface IDeserializer
+    {
+        /// <summary>
+        /// Deserializes the given message
+        /// </summary>
+        /// <param name="input">A stream to read the data to be deserialized</param>
+        /// <param name="type">The type to be created</param>
+        /// <param name="context">An object containing metadata</param>
+        /// <returns>The deserialized message</returns>
+        Task<object> DeserializeAsync(Stream input, Type type, ISerializerContext context);
```

Comment:
It's a little weird that we are separating serialization and deserialization concepts, but we keep ISerializerContext reference in a deserializer context.
I know that segregating into ISerializerContext and IDeserializerContext can also be a little overkill.

Just raising the question... can we live with this, or is it worth addressing?

Reply:
IMO, leaving this as it is seems to be an incomplete job, therefore I think we should address this question now, which may implicate in breaking changes (already in the scope of Kafka Flow v3).

Reply:
Looking at the Confluent.Kafka API, they do not distinguish between Serialization and Deserialization Context https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/IAsyncDeserializer.cs#L44.
Since we're also not exposing this in our APIs in think it is ok to keep a single SerializationContext class.

Reply:
Yup, I have no issues with that. 
For me we can resolve the question and what about you @brmagadutra ?

Reply:
No problem for me, just need to have that in mind to not have this same conversation after. @joelfoliveira and @jose-sousa-8 make sense to add this justification as a comment in the code?

Reply:
Can I resolve this discussion @brmagadutra ?